src/types/Style.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+import type {Theme} from '../theme-provider'
+import type {CSSProperties} from './CSSProperties'
+
+type StyleFunction = (
+  {[key: string]: any}, // props
```

Comment:
One key we _know_ this will always have is the `theme` key which is the `Theme` type.